tests/analysis/test_analysis.py

```diff
@@ -56,6 +57,38 @@ def test_ttest(analysis_df_diff):
     assert 0.05 >= analyser.get_pvalue(analysis_df_diff) >= 0
 
 
+def test_paired_ttest():
+    "This test make sure that pivot table works as expected (inside get_pvalue) and that paired t test returns a possible value"
+
+    analysis_df = pd.DataFrame(
+        {
+            "cluster": ["ES"] * 4 + ["IT"] * 4 + ["PL"] * 4 + ["RO"] * 4,
+            "date": ["2022-01-01", "2022-01-02", "2022-01-03", "2022-01-04"] * 4,
+            "treatment": ["A", "B"] * 8,
+            "target": [0.01] * 16,
+        }
+    )
+    # Changing just one observation so we have a p value
+    analysis_df.loc[1, "target"] = 0.001
+
+    analyser = PairedTTestClusteredAnalysis(
+        cluster_cols=["cluster", "date"], strata_cols=["cluster"]
+    )
+
+    assert 1 >= analyser.get_pvalue(analysis_df) >= 0, "p value is null or inf"
+
+
+def test_paired_ttest_preprocessing():
+
+    analyser = PairedTTestClusteredAnalysis(
+        cluster_cols=["country_code", "city_code"], strata_cols=["country_code"]
+    )
+
+    df_pivot = analyser.preprocessing(df=generate_clustered_data())
+
+    assert df_pivot.isna().sum().sum() == 0, "Unexpected nas in pivot"
```

Comment:
do you think we can make more extensive cheks in here? like, we have a pretty specific image of what should df_pivot be right? some things like:
* number of rows
* average target per country in A and B (we should change the target in the example df for it not to be trivial)

I think it is worth be super sure of what this is doing by more unit testing